### PR TITLE
OPENEUROPA-1140: Add pre and post install perms commands.

### PR DESCRIPTION
**Pull request by richardcanoe**

### CODE REVIEW

File: src/Commands/DrupalCommands.php [hunk 1]
```diff
@@ -150,8 +151,9 @@ public function siteInstall(array $options = [
             ->siteProfile($options['site-profile']);
 
         return $this->collectionBuilder()->addTaskList([
-            $this->sitePreInstall(),
```
Comment (dxvargas) [inline src/Commands/DrupalCommands.php]: We'll still need to call this method.

File: src/Commands/DrupalCommands.php [hunk 2]
```diff
@@ -106,7 +107,7 @@ public function validateSiteInstall(CommandData $commandData)
      * @return \Robo\Collection\CollectionBuilder
      */
     public function siteInstall(array $options = [
-        'root' => InputOption::VALUE_REQUIRED,
+        'root'  => InputOption::VALUE_REQUIRED,
```
Comment (dxvargas) [inline src/Commands/DrupalCommands.php]: One space too much.

**Comment by ademarco** [comment]
We would also need to consider these changes below, as discussed with @richardcanoe, and add tests:

```diff
diff --git a/config/commands/drupal.yml b/config/commands/drupal.yml
index 4d72932..10df396 100644
--- a/config/commands/drupal.yml
+++ b/config/commands/drupal.yml
@@ -27,3 +27,9 @@ command:
     drush-setup:
       options:
         config-dir: ${drupal.root}/drush
+    setup-pre-install-permissions:
+      options:
+        sites-subdir: ${drupal.site.sites_subdir}
+    setup-post-install-permissions:
+      options:
+        sites-subdir: ${drupal.site.sites_subdir}
diff --git a/src/Commands/DrupalCommands.php b/src/Commands/DrupalCommands.php
index 2d636da..3b6f496 100644
--- a/src/Commands/DrupalCommands.php
+++ b/src/Commands/DrupalCommands.php
@@ -304,6 +304,7 @@ class DrupalCommands extends AbstractCommands implements FilesystemAwareInterfac
      */
     public function setupPreInstallPermissions(array $options = [
         'root' => InputOption::VALUE_REQUIRED,
+        'site-subdir' => InputOption::VALUE_REQUIRED,
     ])
     {
         $root = $options['root'];
@@ -327,7 +328,10 @@ class DrupalCommands extends AbstractCommands implements FilesystemAwareInterfac
      * @return \Robo\Collection\CollectionBuilder
      *
      */
-    public function setupPostInstallPermissions()
+    public function setupPostInstallPermissions(array $options = [
+        'root' => InputOption::VALUE_REQUIRED,
+        'site-subdir' => InputOption::VALUE_REQUIRED,
+    ])
     {
         $root = $this->getConfig()->get('drupal.root');
 
```

**Comment by drupol** [comment]
git fetch --all
git rebase origin/master
git push origin --force-with-lease

**Comment by drupol** [comment]
:heart:

**Pull request merged**